clients/src/main/java/org/apache/kafka/clients/consumer/internals/RequestFuture.java

```diff
@@ -128,26 +126,34 @@ public void raise(Errors error) {
         raise(error.exception());
     }
 
-    private void fireSuccess() {
-        for (RequestFutureListener<T> listener : listeners)
+    private void fireSuccess(T value) {
+        while (true) {
+            RequestFutureListener<T> listener = listeners.poll();
+            if (listener == null)
+                break;
             listener.onSuccess(value);
+        }
```

Comment:
Depends on your taste I guess. I don't have a strong preference, but I think the long `for` expression is more difficult to read.